src/librustc/error_codes.rs

```diff
@@ -2184,6 +2184,23 @@ Examples of erroneous code:
 static X: u32 = 42;
 ```
 "##,
+
+E0734: r##"
+A stability attribute has been used outside of the standard library.
+
+Erroneous code examples:
+
+```compile_fail,E0734
+#[rustc_deprecated(since = "b", reason = "text")] // invalid
+#[stable(feature = "a", since = "b")] // invalid
+#[unstable(feature = "b", issue = "0")] // invalid
+fn foo(){}
+```
+
+As the error message states, you cannot use those attributes outside of the
+standard library.
```

Comment:
"These attributes are meant to only be used by the standard library and are rejected in your own crates." Or something like that. If the error is already clear enough then this should explain why, not just restate the error message.